report/4.Solution/ParserImplementation.tex

```diff
@@ -47,45 +48,75 @@ \subsection{Implementing a recursive decent parser}
 		\item Non-terminal Expression
 	\end{itemize}
 \end{itemize}
+\textbf{Deciding what to parse}\\
+The sealed classes, along with the pattern matching of Kotlin, are excellent tools in deciding what should be parsed. 
+In snippet \ref{lis:parseCommand} a subset of the parse command method is shown. 
 
-Based on this, the \textit{parseDeclaration} is implemented as seen in snippet \ref{lis:parseDeclaration}.
+\begin{lstlisting}[language=Kotlin,label=lis:parseCommand,caption=A simplified version of the parse declaration method from the parser.]
+private fun parseCommand(): AstNode.Command {
+val command = when (current.token) {
+is Token.Type -> parseDeclaration()
+is Token.Identifier ->
+if (peek().token == Token.SpecialChar.Equals) parseAssignment() 
+else parseExpression()
+
+...
+
+flushNewLine()
+return command
+}
+\end{lstlisting}
+Snippet \ref{lis:parseCommand} uses the functional functionality of Kotlin, to set the val\footnote{val is a read only variable} \textit{\textbf{command}} to a specific val based on the type of the current token (\textit{\textbf{current.token}}). 
+This method calls other methods based on the type and based on specific rules in the HCL language. 
+For instance all statements starting with a "type" is a declaration, as seen in the CFG, therefore the \textit{\textbf{parseDeclaration}} method is called, and the returned val of this method is then going to be assigned to the command val.
+	
+To simplify the parser and increase readability, certain helper methods is utilized. 
```

Comment:
Overvej at ændre denne sætning til
"... certain helper methods have been created."
eller lign.

Lige meget hvad er "is utilized" forkert.

Reply:
Fixed